cou/utils/juju_utils.py

```diff
@@ -766,3 +760,22 @@ async def get_application_status(self, app_name: str) -> ApplicationStatus:
             if name == app_name:
                 return app
         raise ApplicationNotFound(f"Cannot find '{app_name}' in model '{self.name}'.")
+
+
+async def get_applications_by_charm_name(
```

Comment:
**nitpick (blocking):** 

This function don't have to be async.

Reply:
Fixed